subsys/nrf_security/src/drivers/cracen/cracenpsa/src/platform_keys/platform_keys.c

```diff
@@ -292,17 +294,36 @@ static key_type find_key(uint32_t id, platform_key *key)
 /**
  * @brief Checks whether key usage from a certain domain can access key.
  *
+ * @param[in] domain_id     Originator domain of the key access request.
+ * @param[in] key_id        Key id with domain in N5-N4.
+ *
  * @return psa_status_t
  */
 static psa_status_t verify_access(uint32_t domain_id, uint32_t key_id)
 {
 	switch (PLATFORM_KEY_GET_ACCESS(key_id)) {
 	case ACCESS_INTERNAL:
-		/* Key can be accessed by secure domain only. */
+		/* These keys can be accessed by secure domain. */
 		if (domain_id == DOMAIN_SECURE) {
 			return PSA_SUCCESS;
 		}
-		return PSA_ERROR_NOT_PERMITTED;
+
+		/* Check if access to the target domain key is allowed depending on LCS */
+		enum lcs_domain_id domain_lcs = 0;
+		int status = nrf_domain_to_lcs_domain(PLATFORM_KEY_GET_DOMAIN(key_id), &domain_lcs);
+
+		if (status != 0) {
+			return PSA_ERROR_NOT_PERMITTED;
+		}
+
+		switch (lcs_get(domain_lcs)) {
```

Comment:
If the domain is `SECURE` then this check should not be done. That way it will fail if the OEM keys are already installed.